mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -0,0 +1,488 @@
+from mvpa2.base.progress import ProgressBar
+import os
+from tempfile import mktemp
+from mvpa2.base import externals, warning
+from mvpa2.support import copy
+if externals.exists('h5py'):
+    from mvpa2.base.hdf5 import h5save, h5load
+
+if externals.exists('scipy'):
+    from scipy.sparse import coo_matrix, csc_matrix
+from mvpa2.featsel.helpers import FractionTailSelector, FixedNElementTailSelector
+
+if __debug__:
+    from mvpa2.base import debug
+    if 'SHPAL' in debug.active:
+        def _shpaldebug(msg):
+            debug('SHPAL', "%s" % msg)
+    else:
+        def _shpaldebug(*args):
+            return None
+else:
+    def _shpaldebug(*args):
+        return None
+
+
+def compute_feature_scores(datasets, exclude_from_model=None):
+    """
+    Takes a list of datasets and computes a magical feature
+    score for each feature in each dataset
+    :ref:`Haxby et al., Neuron (2011) <HGC+11>`
+
+    Parameters
+    ----------
+    datasets : list or tuple of datasets
+
+    exclude_from_model: list of dataset indices that won't participate
+                    in voxel selection of others
+
+    Returns
+    -------
+    list : a list of feature scores; higher the score, better the feature
+
+    NOTE: This function assumes that the datasets are zscored
+    """
+    if exclude_from_model is None:
+        exclude_from_model = []
+    feature_scores = [np.zeros(sd.nfeatures) for sd in datasets]
+    for i, sd in enumerate(datasets):
+        for j, sd2 in enumerate(datasets[i + 1:]):
+            corr_temp = np.dot(sd.samples.T, sd2.samples)
+            if j + i + 1 not in exclude_from_model:
+                feature_scores[i] += np.max(corr_temp, axis=1)
+            if i not in exclude_from_model:
+                feature_scores[j + i + 1] += np.max(corr_temp, axis=0)
+    return feature_scores
+
+
+class HyperalignmentMeasure(Measure):
+    """
+    HyperalignmentMeasure combines feature selection and hyperalignment
+    into a single node. This facilitates it's usage in any searchlight
+    or ROI.
+    """
+    is_trained = True
+
+    def __init__(self, hyperalignment=Hyperalignment(ref_ds=0),
+                 featsel=1.0, full_matrix=True, use_same_features=False,
+                 exclude_from_model=None, dtype='float32', **kwargs):
+        Measure.__init__(self, **kwargs)
+        self.hyperalignment = hyperalignment
+        self.featsel = featsel
+        self.use_same_features = use_same_features
+        self.exclude_from_model = exclude_from_model
+        if self.exclude_from_model is None:
+            self.exclude_from_model = []
+        self.full_matrix = full_matrix
+        self.dtype = dtype
+
+    def _call(self, ds):
+        ref_ds = self.hyperalignment.params.ref_ds
+        nsamples, nfeatures = ds[ref_ds].shape
+        if 'roi_seed' in ds[ref_ds].fa:
+            seed_index = np.where(ds[ref_ds].fa.roi_seed)
+        else:
+            seed_index = None
+            self.full_matrix = True
```

Comment:
Add a check to see if full_matrix is true (the default), and raise an exception otherwise?